shell/platform/darwin/ios/framework/Source/FlutterSpellCheckPlugin.mm

```diff
@@ -0,0 +1,134 @@
+  if (self) {
+    [_methodChannel setMethodCallHandler:^(FlutterMethodCall* call, FlutterResult result) {
+      [self handleMethodCall:call result:result];
+    }];
+    _textChecker = [[[UITextChecker alloc] init] retain];
+  }
+  return self;
+}
+
+- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
+  NSString* method = call.method;
+  id args = call.arguments;
+  if ([method isEqualToString:kInitiateSpellCheck]) {
+    NSString* language = args[0];
+    NSString* text = args[1];
+    NSArray<NSDictionary<NSString*, id>*>* spellCheckResult =
+        [self findAllSpellCheckSuggestionsForText:text inLanguage:language];
+    result(spellCheckResult);
+  }
+}
+
+// Get all the misspelled words and suggestions in the entire String.
+//
+// The result will be formatted as am NSArray.
+// Each item of the array is a representation of a misspelled word and suggestions.
+// The format of each item looks like this:
+// {
+//   @"location": 0,
+//   @"length" : 5,
+//   @"suggestions": [@"suggestion1", @"suggestion2"..]
+// }
+//
+// Returns nil if the language is invalid.
+// Returns an empty array if no spell check suggestions.
+- (NSArray<NSDictionary<NSString*, id>*>*)findAllSpellCheckSuggestionsForText:(NSString*)text
+                                                                   inLanguage:(NSString*)language {
+  if (![UITextChecker.availableLanguages containsObject:language]) {
+    return nil;
+  }
+
+  NSMutableArray<NSDictionary<NSString*, id>*>* allSpellSuggestions =
+      [[[NSMutableArray alloc] init] autorelease];
+
+  NSDictionary<NSString*, id>* nextSpellSuggestion;
+  NSInteger nextOffset = 0;
+  do {
+    nextSpellSuggestion = [self findSpellCheckSuggestionsForText:text
+                                                      inLanguage:language
+                                                  startingOffset:nextOffset];
+    if (nextSpellSuggestion != nil) {
+      [allSpellSuggestions addObject:nextSpellSuggestion];
+      nextOffset = [nextSpellSuggestion[@"location"] integerValue] +
+                   [nextSpellSuggestion[@"length"] integerValue];
+    }
+  } while (nextSpellSuggestion != nil);
+
+  return allSpellSuggestions;
+}
+
+// Get the misspelled word and suggestions.
+//
+// The format of result looks like:
+// {
+//   @"location": 0,
+//   @"length" : 5,
+//   @"suggestions": [@"suggestion1", @"suggestion2"..]
+// }
+//
+// Returns nil if no spell check suggestions.
+- (NSDictionary<NSString*, id>*)findSpellCheckSuggestionsForText:(NSString*)text
+                                                      inLanguage:(NSString*)language
+                                                  startingOffset:(NSInteger)startingOffset {
+  FML_DCHECK([UITextChecker.availableLanguages containsObject:language]);
+  NSRange misspelledRange =
+      [self.textChecker rangeOfMisspelledWordInString:text
+                                                range:NSMakeRange(0, text.length)
```

Comment:
Does this API handle empty/nil strings?

Reply:
It handles empty String, which I added a test for. 
We won't get a nil here as I have added some assert above.